app/helpers/application_helper.rb

```diff
@@ -1,4 +1,13 @@
 # frozen_string_literal: true
 
 module ApplicationHelper
+  def current_order
+    if session[:order_id].nil?
+      order = Order.create!(user_id: current_or_guest_user.id)
+      session[:order_id] = order.id
+    else
+      order = Order.find(session[:order_id])
+    end
+    order
+  end
```

Comment:
Good, but need some error handling.
What if my `order_id` doesn't refer to an actual order in DB ?